overlord/confdbstate/transaction.go

```diff
@@ -82,13 +88,20 @@ type marshalledTransaction struct {
 }
 
 func (t *Transaction) MarshalJSON() ([]byte, error) {
+	deltas := make([]map[string]any, len(t.deltas))
```

Comment:
```suggestion
	deltas := make([]map[string]any, 0, len(t.deltas))
```

I think this is missing here. Probably also suggests a missing test?